File: 2. Sorting Techniques/Sorting II/Python/2. Recursive Bubble Sort.py
# cook your dish here
#Recursive Bubble Sort


def recursiveBubbleSort(arr,n):
    if(n==1):
        return
    for j in range(n-1):
        if(arr[j]>arr[j+1]):
            (arr[j],arr[j+1]) = (arr[j+1],arr[j])

    recursiveBubbleSort(arr,n-1)

n = int(input())
arr = list(map(int,input().strip().split()))
recursiveBubbleSort(arr,n)
for num in arr:
    print(num,end=" ")

File: 2. Sorting Techniques/Sorting II/C++/2. Recursive Bubble Sort.cpp
/*
Recursive Bubble Sort
*/

#include <bits/stdc++.h>
using namespace std;

void swap(int arr[],int i,int j){
    int temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

//Function to sort an array using recursive bubble sort algorithm.
void recursiveBubbleSort(int arr[], int n)
{
    if(n==1){
        return;
    }
    for(int j=0;j<n-1;j++){
        if(arr[j]>arr[j+1]){
            swap(arr,j,j+1);
        }
    }
    recursiveBubbleSort(arr,n-1);
}

int main() {
	// your code goes here
    int n;
    cin >> n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    recursiveBubbleSort(arr,n);
    for(int i=0;i<n;i++){
        cout << arr[i] << " "; 
    }
	return 0;
}
